pkg/ext/delegate_test.go

```diff
@@ -0,0 +1,1012 @@
+package ext
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	gomock "go.uber.org/mock/gomock"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+func TestDelegate_Watch(t *testing.T) {
+	type input struct {
+		ctx             context.Context
+		internaloptions *metainternalversion.ListOptions
+	}
+
+	type output struct {
+		watch watch.Interface
+		err   error
+	}
+
+	type testCase struct {
+		name                    string
+		input                   input
+		expected                output
+		storeSetup              func(*MockStore[*TestType, *TestTypeList])
+		simulateConvertionError bool
```

Comment:
typo: should be `simulateConversionError`

Reply:
I'm sorry. Just fixed it. Thanks!